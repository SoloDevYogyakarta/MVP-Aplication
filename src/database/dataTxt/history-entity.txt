{"id":13,"type":"Tuner Up","date":"2023-10-21T13:11:30.209Z","product_id":"jdJ_bHVhSPV35O5IgcnDn","public_id":"rPJ0T8yQcUy2dLlGFYF9P"}