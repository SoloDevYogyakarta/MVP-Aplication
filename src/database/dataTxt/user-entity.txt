{"id":1088,"public_id":"io7NL8MF7mPHdla16p3X0","username":"updated-Opal_Rempel28","email":"Dameon_Cronin16@gmail.com","phone_number":null,"password":"$2b$10$ix8JcUoVPanbpNiAczvphut0WyjV/wcUUGw0/WpCedOFk.dkqSpvm","createdAt":"2023-10-21T12:41:46.989Z","updatedAt":"2023-10-21T12:41:47.100Z"}