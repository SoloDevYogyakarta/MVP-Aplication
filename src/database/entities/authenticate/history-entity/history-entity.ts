import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class UserHistoryEntity extends BaseEntity {
  plat_number: string;
  phone_number: string;
  name: string;
  address: string;
  user_id: string;
}

export type UserHistoryCreationAttribute = Optional<UserHistoryEntity, 'id'>;
export interface UserHistoryInstance
  extends Model<UserHistoryCreationAttribute, UserHistoryEntity>,
    UserHistoryEntity {}
export const userHistoryEntity = sequelize.define<UserHistoryInstance>(
  'HISTORY',
  {
    public_id: DataTypes.STRING,
    plat_number: DataTypes.STRING,
    phone_number: DataTypes.STRING,
    name: DataTypes.STRING,
    address: DataTypes.STRING,
    user_id: DataTypes.STRING,
  } as ModelAttributes<UserHistoryInstance>,
  {
    tableName: 'HISTORY',
    schema: 'AUTHENTICATE',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
