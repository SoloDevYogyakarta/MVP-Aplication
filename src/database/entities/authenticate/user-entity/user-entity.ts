import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { SuperTest } from 'supertest';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class UserEntity extends BaseEntity {
  plat_number: string;
  phone_number: string;
  name: string;
  motor: string;
  year_production: number;
  address: string;
  role: string;
  password: string;
}

export type UserCreationAttribute = Optional<UserEntity, 'id'>;
export interface UserInstance
  extends Model<UserCreationAttribute, UserEntity>,
    UserEntity {}
export const userEntity = sequelize.define<UserInstance>(
  'USER',
  {
    plat_number: DataTypes.STRING,
    phone_number: DataTypes.STRING,
    name: DataTypes.STRING,
    motor: DataTypes.STRING,
    year_production: DataTypes.INTEGER,
    address: DataTypes.STRING,
    role: DataTypes.STRING,
    password: DataTypes.STRING,
  } as ModelAttributes<UserInstance>,
  {
    tableName: 'USER',
    schema: 'AUTHENTICATE',
  },
);
