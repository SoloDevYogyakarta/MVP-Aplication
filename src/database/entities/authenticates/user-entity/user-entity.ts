import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { nanoid } from 'nanoid';
import bcrypt from 'bcrypt';
import { sequelize } from '../../entity';
import { BaseEntity } from '../../base-entity/base-entity';

export class UserEntity extends BaseEntity {
  username: string;
  email: string;
  phone_number: number;
  password: string;
  role: string;
  file_id: string;
}

export type UserCreationAttribute = Optional<UserEntity, 'id'>;
export interface UserInstance
  extends Model<UserCreationAttribute, UserEntity>,
    UserEntity {}
export const userEntity = sequelize.define<UserInstance>(
  'USER',
  {
    public_id: DataTypes.STRING,
    username: DataTypes.STRING,
    email: DataTypes.STRING,
    phone_number: DataTypes.INTEGER,
    role: DataTypes.STRING,
    password: DataTypes.STRING,
    file_id: DataTypes.STRING,
  } as ModelAttributes<UserInstance>,
  {
    tableName: 'USER',
    schema: 'AUTHENTICATE',
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
        instance.password = bcrypt.hashSync(instance.password, 10);
      },
    },
  },
);
