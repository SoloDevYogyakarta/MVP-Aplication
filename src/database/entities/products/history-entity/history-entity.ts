import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class HistoryEntity extends BaseEntity {
  type: string;
  date: Date;
  product_id: string;
  user_id: string;
}

export type HistoryCreationAttribute = Optional<HistoryEntity, 'id'>;
export interface HistoryInstance
  extends Model<HistoryCreationAttribute, HistoryEntity>,
    HistoryEntity {}
export const historyEntity = sequelize.define<HistoryInstance>(
  'HISTORY',
  {
    public_id: DataTypes.STRING,
    type: DataTypes.STRING,
    date: DataTypes.DATE,
    product_id: DataTypes.STRING,
    user_id: DataTypes.STRING,
  } as ModelAttributes<HistoryInstance>,
  {
    tableName: 'HISTORY',
    schema: 'PRODUCTS',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
