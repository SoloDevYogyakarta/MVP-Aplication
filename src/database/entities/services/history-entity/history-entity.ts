import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class HistoryEntity extends BaseEntity {
  name: string;
  title: string;
  desc: string;
  price: number;
  user_id: string;
}

export type HistoryCreationAttribute = Optional<HistoryEntity, 'id'>;
export interface HistoryInstance
  extends Model<HistoryCreationAttribute, HistoryEntity>,
    HistoryEntity {}
export const historyEntity = sequelize.define<HistoryInstance>(
  'HISTORY',
  {
    public_id: DataTypes.STRING,
    name: DataTypes.STRING,
    title: DataTypes.STRING,
    desc: DataTypes.STRING,
    price: DataTypes.INTEGER,
    user_id: DataTypes.STRING,
  } as ModelAttributes<HistoryInstance>,
  {
    tableName: 'HISTORY',
    schema: 'SERVICES',
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
