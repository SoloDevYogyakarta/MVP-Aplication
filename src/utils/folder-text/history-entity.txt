{"id":16,"name":"Handmade Wooden Mouse","title":"Bronze","desc":"The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design","price":69653,"order_id":8,"updatedAt":"2023-11-07T07:15:16.412Z","createdAt":"2023-11-07T07:15:16.412Z"}