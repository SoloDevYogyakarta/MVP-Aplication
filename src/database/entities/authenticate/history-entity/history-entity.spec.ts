import { ModelCtor } from 'sequelize';
import { omit } from 'lodash';
import { getfield } from '../../../../utils/get-field/get-field';
import { UserInstance } from '../user-entity/user-entity';
import { userHistoryEntity, UserHistoryInstance } from './history-entity';
import { createpath } from '../../../../utils/system/system';

describe('UserHistoryEntity', () => {
  let user!: UserInstance;
  let destroy!: UserInstance;
  let public_id!: string;
  let entity: ModelCtor<UserHistoryInstance>;

  beforeEach(() => {
    entity = userHistoryEntity;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  try {
    user = getfield('user-http-entity');
  } catch (err) {
    // empty
  }
  try {
    public_id = getfield('user-history-entity').public_id;
  } catch (err) {
    // empty
  }
  try {
    destroy = getfield('user-destroy-with-relationship-entity');
  } catch (err) {
    // empty
  }

  if (user) {
    it('create', async () => {
      const create = await entity.create({
        ...omit(user, [
          'id',
          'public_id',
          'motor',
          'year_production',
          'role',
          'password',
          'createdAt',
          'updatedAt',
        ]),
        user_id: user.public_id,
      });
      create.save();
      createpath('../folder-text/user-history-entity.txt', create);
      expect(create.user_id).toEqual(user.public_id);
    });
  }

  if (destroy) {
    it('create', async () => {
      const create = await entity.create({
        ...omit(destroy, [
          'id',
          'public_id',
          'motor',
          'year_production',
          'role',
          'password',
          'createdAt',
          'updatedAt',
        ]),
        user_id: destroy.public_id,
      });
      create.save();
      createpath('../folder-text/user-history-destroy-entity.txt', create);
      expect(create.user_id).toEqual(destroy.public_id);
    });
  }

  if (public_id) {
    it('findOne', async () => {
      const findOne = await entity.findOne({ where: { public_id } });
      expect(findOne.public_id).toEqual(public_id);
    });

    it('findAll', async () => {
      const findAll = await entity.findAll();
      expect(findAll.length).not.toEqual(0);
    });
  }
});
