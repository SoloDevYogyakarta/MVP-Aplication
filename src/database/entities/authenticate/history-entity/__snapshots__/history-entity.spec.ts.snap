// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHistoryEntity render correctly 1`] = `[Function]`;
